metaseq/modules/feedforward_network.py

```diff
@@ -34,16 +34,18 @@ def FeedForwardNetwork(x, fc1, activation_fn, fc2, dropout_module):
         x = x + bias_fc2
     elif model_parallel:
         # here, we do the bias computation inside fc1 and fc2 AND gather_output
-        x, _ = fc1(x)
-        x = activation_fn(x)
+        fc1_out, _ = fc1(x)
+        x = activation_fn(x, fc1_out, True)
+        del fc1_out
         x, _ = fc2(x)
-    elif has_fused_bias_gelu and activation_fn == gelu:
+    elif has_fused_bias_gelu and activation_fn.fn == gelu:
         x = F.linear(x, fc1.weight, None)
         x = fused_bias_gelu(x, fc1.bias)
         x = F.linear(x, fc2.weight, fc2.bias)
     else:
-        x = fc1(x)
-        x = activation_fn(x)
+        fc1_out = fc1(x)
+        x = activation_fn(x, fc1_out, False)
+        del fc1_out
         x = fc2(x)
```

Comment:
Instead of doing the manual deletion, let's just inline the call so it goes out of scope immediately

Reply:
I guess that's awkward with the tuple. What is the second return value anyway?

Reply:
It's the bias, see line 30. When skip_bias_add = True (which we set only when we want to use fused gelu), then the bias is not added to the weights and instead returned separately. In this case here, given we are adding bias the second return value is just None. Maybe there's a smart way to skip this